### PR TITLE
[IIIF-769] Migrate S3 states

### PR DESCRIPTION
**Pull request by cachemeoutside**
This PR contains terraform configs to create a prod and test S3 statefile stored in S3. The existing prod, stage, and test buckets have been migrated to the new statefiles.

### CODE REVIEW

File: environments/prod/s3/prod.tfvars [hunk 1]
```diff
@@ -0,0 +1,12 @@
+prod-iiif_cantaloupe_s3_source_bucket = "prod-cantaloupe-iiif-source"
+prod-iiif_cantaloupe_s3_cache_bucket = "prod-iiif-cantaloupe-cache"
+prod-iiif_fester_s3_source_bucket = "prod-iiif-fester-source"
+
+stage-iiif_cantaloupe_s3_source_bucket = "prod-cantaloupe-iiif-source"
```
Comment (ksclarke) [inline environments/prod/s3/prod.tfvars:5]: Should this value be stage-cantaloupe-iiif-source?

File: environments/prod/s3/variables.tf [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  type = string
+  default = "cantaloupesourcebucket"
+}
+
+variable "prod-iiif_fester_s3_source_bucket" {
+  type = string
+  default = "festersourcebucket"
+}
+
+variable "stage-iiif_cantaloupe_s3_cache_bucket" {
```
Comment (ksclarke) [inline environments/prod/s3/variables.tf:21]: I may not understand the structure, but should stage variables be in the `environments/prod` dir?

File: environments/test/s3/terraform.tf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+terraform {
+  required_version = "~> 0.12.20"
+  backend "s3" {
+    encrypt = true
+    bucket = "prod-services-terraform-state"
```
Comment (ksclarke) [inline environments/test/s3/terraform.tf:5]: And again I may not get the arrangement, but a prod bucket is being referenced from an `environments/test` file?

File: environments/prod/s3/prod.tfvars (hunk 1 above)
Comment (cachemeoutside) [reply environments/prod/s3/prod.tfvars]: The stage bucket is differently. Its reading from the prod bucket. There actually isn't a stage cantaloupe s3 source bucket and the newest commits reflects that change.

File: environments/prod/s3/variables.tf (hunk 2 above)
Comment (cachemeoutside) [reply environments/prod/s3/variables.tf]: Yes, good call here. I moved all stage resources to its own `environments/stage` directory in this latest commit.

File: environments/test/s3/terraform.tf (hunk 3 above)
Comment (cachemeoutside) [reply environments/test/s3/terraform.tf:5]: This is expected. I only configured one bucket to store our Terraform states. The identifier between test/stage/prod environment has the environment prefixed to its key name.

**Comment by cachemeoutside** [comment]
Ready for review again.

**Pull request merged**